mlflow/tracing/provider.py

```diff
@@ -118,7 +119,15 @@ def _setup_tracer_provider(disabled=False):
         _MLFLOW_TRACER_PROVIDER = trace.NoOpTracerProvider()
         return
 
-    if is_in_databricks_model_serving_environment():
+    if should_use_otlp_exporter():
+        # Export to OpenTelemetry Collector when configured
+        from mlflow.tracing.processor.otel import OtelSpanProcessor
+
+        exporter = get_otlp_exporter()
+        processor = OtelSpanProcessor(exporter)
+
+    elif is_in_databricks_model_serving_environment():
+        # Export to Inference Table when running in Databricks Model Serving
```

Comment:
Can we alert users if the ENV VAR is set for an OTEL collector when they're in Databricks Model Serving why this configuration isn't valid? I'm sure that many users will attempt to set this up and the logic here will pre-emptively attempt to use that configuration (which won't work due to restrictions within dblet node initialization rulesets). Can we modify `should_use_otlp_exporter()` so that it returns False if in model serving?

Reply:
Hmm why does it considered invalid? Does model serving container prohibit outbound traffic (which I believe not...?).

Reply:
Discussed offline :) Provided that the functionality works and doesn't exhibit some port access error, this is not a blocker. We can add a note in the docs for Databricks users who opt-in for this that they will miss out on integration functionality if they configure this.